### ISSUE TITLE
Convolutional neural network with LSTM

### ISSUE THREAD

**Issue by tjdurant** [keras-team/keras#8861]
Hello,

I am working on a project where we want to implement and autoencoder approach to an image classification problem. The approach we are looking to take would be similar to that described here: [https://arxiv.org/pdf/1710.10501.pdf](url)

I would like to ask if this type of model is possible in Keras. Currently I am struggling with the connection point between the Dense CNN and the LSTM. 

```
def AutoEncoder(input_shape=None, depth=40, nb_dense_block=3, growth_rate=12, nb_filter=-1, nb_layers_per_block=-1,
                 bottleneck=False, reduction=0.0, dropout_rate=0.0, weight_decay=1e-4, subsample_initial_block=False,
                 include_top=True, weights=None, input_tensor=None,classes=10, activation='softmax',own_data=None):
    
    # Determine proper input shape
    input_shape = _obtain_input_shape(input_shape,
                                      default_size=32,
                                      min_size=8,
                                      data_format=K.image_data_format(),
                                      require_flatten=include_top)

    if input_tensor is None:
        img_input = Input(shape=input_shape)
    else:
        if not K.is_keras_tensor(input_tensor):
            img_input = Input(tensor=input_tensor, shape=input_shape)
        else:
            img_input = input_tensor

    x = __create_dense_net(classes, img_input, include_top, depth, nb_dense_block,
                           growth_rate, nb_filter, nb_layers_per_block, bottleneck, reduction,
                           dropout_rate, weight_decay, subsample_initial_block, activation)

    # Ensure that the model takes into account
    # any potential predecessors of `input_tensor`.
    if input_tensor is not None:
        inputs = get_source_inputs(input_tensor)
    else:
        inputs = img_input

#     x = LSTM? 
    
    autoencoder = Model(img_input,x)
    
    return autoencoder
```

Which layers would be used from the Keras library and how would they be used to accept the input from the last layer of the Dense CNN encoder depicted here:

concatenate_922 (Concatenate)   (None, 7, 7, 292)    0   concatenate_921[0][0]   conv2d_1990[0][0]                
batch_normalization_1984 (Batch (None, 7, 7, 292)    1168        concatenate_922[0][0]            
activation_1983 (Activation)    (None, 7, 7, 292)    0           batch_normalization_1984[0][0]   
max_pooling2d_52 (MaxPooling2D) (None, 3, 3, 292)    0           activation_1983[0][0]            
Total params: 435,878
Trainable params: 428,230
Non-trainable params: 7,648

Any help would be greatly appreciated or please let me know what additional information would be needed.

Thank you in advance,

**Comment by gabrieldemarmiesse** [comment]
Hi!
I read quickly the paper mentioned. Was it voluntary to exclude the flatten layer at the end of the dense cnn? 
What you should do is using the flatten layer at the end of your cnn. If you do that, your resulting tensor will be 2D. After that, you can use the the keras RepeatVector to end up with a 3D tensor (as done in the paper). But before giving that to the LSTM, you must concatenate this 3D tensor with the labels like done in the paper (keras layer concatenate, with axis=2).

I hope it helps.
